src/GovUk.Education.ExploreEducationStatistics.Admin.Tests/Services/DataSetVersionServiceTests.cs

```diff
@@ -96,12 +100,155 @@ await TestApp.AddTestData<ContentDbContext>(context =>
+
+            DataSetVersion dataSetVersion2 = DataFixture
+                .DefaultDataSetVersion()
+                .WithDataSet(DataFixture.DefaultDataSet())
+                .WithRelease(DataFixture.DefaultDataSetVersionRelease()
+                    .WithReleaseFileId(releaseDataFile2.Id)
+                    .WithSlug(releaseVersion.Slug)
+                    .WithTitle(releaseVersion.Title));
+
+            await TestApp.AddTestData<PublicDataDbContext>(context =>
+            {
+                context.DataSetVersions.AddRange(dataSetVersion1, dataSetVersion2);
+            });
+
+            var service = TestApp.Services.GetRequiredService<IDataSetVersionService>();
+
+            const string updatedSlug = "2024-25";
+            const string updatedTitle = "Academic year 2024/25";
+
+            await service.UpdateVersionsForReleaseVersion(
+                releaseVersion.Id,
+                slug: updatedSlug,
+                title: updatedTitle);
+
+            await using var publicDataDbContext = TestApp.GetDbContext<PublicDataDbContext>();
+
+            var actualDataSetVersions = await publicDataDbContext.DataSetVersions
+                .AsNoTracking()
+                .ToListAsync();
+
+            Assert.Equal(2, actualDataSetVersions.Count);
+
+            Assert.All(actualDataSetVersions,
+                dataSetVersion =>
+                {
+                    Assert.Equal(updatedSlug, dataSetVersion.Release.Slug);
+                    Assert.Equal(updatedTitle, dataSetVersion.Release.Title);
+                });
+        }
+
+        [Fact]
+        public async Task UnrelatedDataSetVersionsAreNotUpdated()
+        {
+            var (releaseVersion1, releaseVersion2) = DataFixture
+                .DefaultReleaseVersion()
+                .Generate(2)
+                .ToTuple2();
+
+            ReleaseFile releaseVersion1DataFile = DataFixture
+                .DefaultReleaseFile()
+                .WithFile(DataFixture.DefaultFile(FileType.Data))
+                .WithReleaseVersion(releaseVersion1);
+
+            ReleaseFile releaseVersion2DataFile = DataFixture
+                .DefaultReleaseFile()
+                .WithFile(DataFixture.DefaultFile(FileType.Data))
+                .WithReleaseVersion(releaseVersion2);
+
+            await TestApp.AddTestData<ContentDbContext>(context =>
+            {
+                context.ReleaseVersions.AddRange(releaseVersion1, releaseVersion2);
+                context.ReleaseFiles.AddRange(releaseVersion1DataFile, releaseVersion2DataFile);
+            });
+
+            DataSetVersion dataSetVersion1 = DataFixture
+                .DefaultDataSetVersion()
+                .WithDataSet(DataFixture.DefaultDataSet())
+                .WithRelease(DataFixture.DefaultDataSetVersionRelease()
+                    .WithReleaseFileId(releaseVersion1DataFile.Id)
+                    .WithSlug(releaseVersion1.Slug)
+                    .WithTitle(releaseVersion2.Title));
```

Comment:
think this should be `WithTitle(releaseVersion1.Title)`

Reply:
Good spot! Fixed now, thanks. 🙂